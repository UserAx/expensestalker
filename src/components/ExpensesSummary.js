import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formatedTotals = numeral(expensesTotal/100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h3 className="page-header-title">Viewing <span>{expensesCount}</span> {expenseWord}, 
                a total of <span>{formatedTotals}</span>. </h3>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

//we need this because our expenses exist in the redux store state object.
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);