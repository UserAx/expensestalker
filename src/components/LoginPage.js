import React from 'react';
import {startLogin} from '../actions/auth';
import {connect } from 'react-redux';

export const LoginPage = ({startLogin}) => (
    <div className ="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify App</h1>
            <p>It is time to get your expenses under control.</p>
            <button className="button" onClick={startLogin}>Log In</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);