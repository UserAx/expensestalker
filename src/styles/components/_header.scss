.header {
    background: $dark-blue;
}

.header__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $s-size 0;
}

.header__title{
    color: white;
    text-decoration: none;
    h1 {
        margin: 0
    }
}

.content-container{
    margin: 0 auto;
    padding: 0 $m-size;
    max-width: 80rem;
}