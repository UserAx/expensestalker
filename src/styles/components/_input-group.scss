.input-group {
    display: flex;
    margin-bottom: $m-size;
    flex-direction: column;
    @media (min-width: $desktop-breakpoint) {//when the minimum size becomes the one of a desktop
        flex-direction: row;
        margin-bottom: $l-size;
    }
}

.input-group__item{
    margin: 0 $s-size 0 0;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $s-size;
    }
}