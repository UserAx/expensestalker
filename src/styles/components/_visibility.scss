.show-for-mobile {
    @media(min-width: $desktop-breakpoint) {//when the minimum size becomes the one of a desktop
        display: none;
    }
}

.show-for-desktop {
    @media(max-width: $desktop-breakpoint - .01rem) {//when the maximum size becomes the one of a mobile
        display: none;
    }
}