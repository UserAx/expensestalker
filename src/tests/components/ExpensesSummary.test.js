import React from 'react';
import { shallow } from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render expenses summary with a single expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal= {235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses summary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal= {23500}/>);
    expect(wrapper).toMatchSnapshot();
});

