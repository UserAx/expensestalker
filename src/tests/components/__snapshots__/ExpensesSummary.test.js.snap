// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary with a single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h3
      className="page-header-title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
      , a total of 
      <span>
        $2.35
      </span>
      . 
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render expenses summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h3
      className="page-header-title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
      , a total of 
      <span>
        $235.00
      </span>
      . 
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
